Fix TODO

YanShuq1 opened the pull request:
# feat(route):完成TODO项的页面跳转功能  
用户登录凭证过期或被踢下线时返回登录页，使用了GlobalKey进行无context跳转。

YanShuq1 committed:
feat(route):完成TODO项的页面跳转功能
  用户登录凭证过期或被踢下线时返回登录页，使用了GlobalKey进行无context跳转。

xhzq233 commented on lib/im.dart:
```diff
@@ -49,13 +57,15 @@ Future<void> initIM() async {
         SmartDialog.showToast('IM当前用户被踢下线');
-        // TODO: pop to login page
+        // Done: pop to login page(无context跳转)
+        navigatorKey.currentState?.push(LoginPage.route());
```
這裏應該是pop，比如當前用戶在聊天，應該登出后不允許該用戶繼續聊天，要把該頁面pop出去

xhzq233 replied:
修完就把TODO直接刪了就行0.0

YanShuq1 replied:
好嘟O.O

xhzq233 replied:
萬一當前頁面有多個呢，只pop一個是不是不行

YanShuq1 replied:
那我看看O.o

YanShuq1 replied:
那用pushNamedAndRemoveUntil方法可以做到但是要注册路由名字诶

xhzq233 replied:
既然有pushnamed，那肯定有push，你可以再看一看

xhzq233 replied:
你的format注意下

YanShuq1 replied:
什么format有问题O.o

YanShuq1 committed:
feat(route):完成TODO项的页面跳转功能
  用户登录凭证过期或被踢下线时返回登录页，使用了GlobalKey进行无context跳转。

YanShuq1 committed:
Merge branch 'xhzq233:main' into fix_todo

YanShuq1 committed:
feat(route):完成TODO项的页面跳转功能
  用户登录凭证过期或被踢下线时返回登录页，使用了GlobalKey进行无context跳转。
  fix:将push改成pushNamedAndRemoveUntil方法，保证了路由栈清空登录页前的所有路由。

YanShuq1 committed:
feat(route):完成TODO项的页面跳转功能
  用户登录凭证过期或被踢下线时返回登录页，使用了GlobalKey进行无context跳转。
  将push方法改为pushAndRemoveUntil方法，跳转之前清空了路由栈。

YanShuq1 committed:
feat(route):完成TODO项的页面跳转功能
  用户登录凭证过期或被踢下线时返回登录页，使用了GlobalKey进行无context跳转。
  将push方法改为pushAndRemoveUntil方法，跳转之前清空了路由栈。

xhzq233 commented on lib/pages/login/login_page.dart:
commit前检查一下自己改动了什么，是不是有不该改动的

YanShuq1 committed:
feat(route):完成TODO项的页面跳转功能
  用户登录凭证过期或被踢下线时返回登录页，使用了GlobalKey进行无context跳转。
  将push方法改为pushAndRemoveUntil方法，跳转之前清空了路由栈。
  (已格式化)
